lib/time.js

```diff
@@ -349,6 +415,219 @@ function CronTime(luxon) {
 			return date;
 		},
 
+		/**
+		 * Search backwards in time 1 minute at a time, to detect a DST forward jump.
+		 * When the jump is found, the range of the jump is investigated to check for acceptable cron times.
+		 * A pair is returned, whose first is a boolean representing if an acceptable time was found inside the jump,
+		 * and whose second is a DateTime representing the first millisecond after the jump.
+		 *
+		 * The input date is expected to be decently close to a DST jump.
+		 * Up to a day in the past is checked before an error is thrown.
+		 * @param date
+		 */
+		_findPreviousDSTJump: function (date) {
+			/** @type number */
+			let expectedMinute, expectedHour, actualMinute, actualHour;
+			/** @type DateTime */
+			let maybeJumpingPoint;
+
+			// representing one day of backwards checking. If this is hit, the input must be wrong.
+			const iterationLimit = 60 * 24;
+			let iteration = 0;
+			do {
+				if (iteration++ > iterationLimit) {
+					throw new Error(
+						`ERROR: This DST checking related function assumes the input DateTime (${date.toISO()}) is within 24 hours of a DST jump.`
+					);
+				}
+
+				expectedMinute = date.minute - 1;
+				expectedHour = date.hour - 1;
+
+				if (expectedMinute < 0) expectedMinute += 60;
+				if (expectedHour < 0) expectedHour += 60;
```

Comment:
why would the expected hour be 60?

Reply:
see https://github.com/kelektiv/node-cron/pull/639#discussion_r1116354937, this whole loop was broken